packages/strapi-plugin-content-manager/test/content-type-api.test.e2e.js

```diff
@@ -0,0 +1,756 @@
+      const product = {
+        name: 'Product 2',
+        description: 'Product description',
+        published_at: '2020-08-20T10:27:55.866Z',
+      };
+      const res = await rq({
+        method: 'POST',
+        url: '/content-manager/explorer/application::product-with-dp.product-with-dp',
+        body: product,
+      });
+
+      expect(res.statusCode).toBe(200);
+      expect(res.body).toMatchObject(_.omit(product, 'published_at'));
+      expect(res.body.published_at).toBeNull();
+      data.productsWithDP.push(res.body);
+    });
+
+    test('Read all products', async () => {
+      const res = await rq({
+        method: 'GET',
+        url: '/content-manager/explorer/application::product-with-dp.product-with-dp',
+      });
+
+      expect(res.statusCode).toBe(200);
+      expect(Array.isArray(res.body)).toBe(true);
+      expect(res.body).toEqual(
+        expect.arrayContaining([
+          expect.objectContaining({
+            name: 'Product 1',
+            description: 'Product description',
+          }),
+        ])
+      );
+      res.body.forEach(p => {
+        expect(p.published_at).toBeNull();
+      });
+    });
+
+    test('Update a draft', async () => {
+      const product = {
+        name: 'Product 1 updated',
+        description: 'Updated Product description',
+      };
+      const res = await rq({
+        method: 'PUT',
+        url: `/content-manager/explorer/application::product-with-dp.product-with-dp/${data.productsWithDP[0].id}`,
+        body: product,
+      });
+
+      expect(res.statusCode).toBe(200);
+      expect(res.body).toMatchObject(_.omit(product, 'published_at'));
+      expect(res.body.id).toEqual(data.productsWithDP[0].id);
+      expect(res.body.published_at).toBeNull();
+      data.productsWithDP[0] = res.body;
+    });
+
+    test('Update Products + cannot overwrite published_at', async () => {
+      const product = {
+        name: 'Product 1 updated',
+        description: 'Updated Product description',
+      };
+      const res = await rq({
+        method: 'PUT',
+        url: `/content-manager/explorer/application::product-with-dp.product-with-dp/${data.productsWithDP[0].id}`,
+        body: product,
+      });
+
+      expect(res.statusCode).toBe(200);
+      expect(res.body).toMatchObject(product);
+      expect(res.body.id).toEqual(data.productsWithDP[0].id);
+      expect(res.body.published_at).toBeNull();
+      data.productsWithDP[0] = res.body;
+    });
+
+    test('Publish an product, expect published_at to be defined', async () => {
+      const entry = data.productsWithDP[0];
+
+      let { body } = await rq({
+        url: `/content-manager/explorer/application::product-with-dp.product-with-dp/publish/${entry.id}`,
+        method: 'POST',
+      });
+
+      data.productsWithDP[0] = body;
+
+      expect(body.published_at).toEqual(expect.any(String));
+      expect(isNaN(new Date(body.published_at).valueOf())).toBe(false);
```

Comment:
Don't forget to check if published at is not null (otherwise it will pass this expect without failing)

Reply:
In fact it's ok because just above it checks if it's a string, I didn't see during our call